pandas/core/arrays/datetimes.py

```diff
@@ -639,7 +647,7 @@ def astype(self, dtype, copy: bool = True):
     # Rendering Methods
 
     def _format_native_types(
-        self, *, na_rep="NaT", date_format=None, **kwargs
+        self, *, na_rep: object = "NaT", date_format=None, **kwargs
```

Comment:
Curious what are the other `na_rep` besides `str` that make this `object`?

Reply:
It is probably better to stick with `str`, I widened the type because they typically accept anything that can be converted to a string and because here na_rep is a float:
https://github.com/pandas-dev/pandas/blob/12dce19a74b7cd5badad0f61ca079b873c1b6089/pandas/core/arrays/datetimelike.py#L1791

Reply:
I changed it back to `str` and only added `float` whenever it was needed. I also added `NaRep = str` in `pandas._typing` so that the type could easily be changed whenever needed.

Reply:
> Happy to remove `NaRep` and just have `str` everywhere

Yeah I think this would be better. IMO where it is float (the only one being `np.nan` maybe?), it should just be changed to `"nan"` and the caller should pass `str(whatever)`

Reply:
Based on this issue, it seems intended to allow floats https://github.com/pandas-dev/pandas/issues/29578

The failing test is https://github.com/pandas-dev/pandas/blob/65af4ef2043a5f879dc42e009de1323fdcea225b/pandas/tests/arrays/test_datetimelike.py#L941

Reply:
I left `na(t)_rep` as `str` in most places and added `float` when needed (but more functions might in practise accept floats as well)